spec/features/guests/show_spec.rb

```diff
@@ -0,0 +1,63 @@
+require "rails_helper"
+
+RSpec.describe "Guest Show Page" do
+  before(:each) do 
+    @hotel_1 = Hotel.create!(name: "Marriott", location: "Silverthorne")
+    @hotel_2 = Hotel.create!(name: "Hilton", location: "Breckenridge")
+
+    @room_1 = Room.create!(rate: 200, suite: "Presidential", hotel_id: @hotel_1.id)
+    @room_2 = Room.create!(rate: 100, suite: "Deluxe", hotel_id: @hotel_1.id)
+    @room_3 = Room.create!(rate: 90, suite: "Standard", hotel_id: @hotel_2.id)
+
+    @guest_1 = Guest.create!(name: "Michael Scott", nights: 1)
+    @guest_2 = Guest.create!(name: "Dwight Schrute", nights: 2)
+    @guest_3 = Guest.create!(name: "Jim Halpert", nights: 3)
+
+    @guest_room_1 = GuestRoom.create!(guest_id: @guest_1.id, room_id: @room_1.id)
+    @guest_room_2 = GuestRoom.create!(guest_id: @guest_2.id, room_id: @room_1.id)
+    @guest_room_3 = GuestRoom.create!(guest_id: @guest_2.id, room_id: @room_2.id)
+    @guest_room_4 = GuestRoom.create!(guest_id: @guest_3.id, room_id: @room_3.id)
+  end
+
+  it "displays guest name, rooms they've stayed in, and room details" do
+    visit "/guests/#{@guest_2.id}"
+
+    expect(page).to have_content("#{@guest_2.name}")
+
+    within("#rooms") do
+      expect(page).to have_content("#{@guest_2.rooms.first.hotel.name}")
```

Comment:
These assertions are good, but could be more meaningful if you include a text label from the view to indicate they show up in the right place, i.e. assert the presence of "Hotel Name: #{guest_2.rooms.first.hotel.name}" instead of just the name. This is especially helpful with asserting the presence of integer fields, like `rate` because there are often numbers on pages for different reasons (phone numbers, addresses, etc) so you don't want to get a false positive. Better to anchor it to an assertion testing for "Nightly Rate: $300" rather than just "300"